File: assignment_1/Customer.java
// This is the Customer class - it stores details about customers who buy products
public class Customer {
    int cid;            
    int uid;            
    String name;        
    String phone;       
    String email;       
    String address;     
    double totalPurchase; 

    // This function shows all customer details
    // No parameters, returns nothing
    void displayCustomer() {
        System.out.println("Customer ID    : " + cid);
        System.out.println("User ID        : " + uid);
        System.out.println("Name           : " + name);
        System.out.println("Phone          : " + phone);
        System.out.println("Email          : " + email);
        System.out.println("Address        : " + address);
        System.out.println("Total Purchase : Rs." + totalPurchase);
    }

    // This function shows that a customer purchased a product
    // Parameter: p (a Product object) - tells which product was bought
    // Returns nothing
    void purchaseProduct(Product p) {
        System.out.println(name + " purchased " + p.productName);
    }

    // This function shows that a customer returned a product
    // Parameter: p (a Product object) - tells which product was returned
    // Returns nothing
    void returnProduct(Product p) {
        System.out.println(name + " returned " + p.productName);
    }

    // This function shows that a customer replaced a product
    // Parameter: p (a Product object) - tells which product was replaced
    // Returns nothing
    void replaceProduct(Product p) {
        System.out.println(name + " replaced " + p.productName);
    }
}

File: python/customer.py
class Customer:
    def __init__(self):
        self.cid = 0
        self.uid = 0
        self.User_name = ""
        self.contact = ""
        self.email = ""
        self.address = ""
        self.total_purchase = 0.0

    def displayCustomer(self):
        print("Customer ID    :", self.cid)
        print("User ID        :", self.uid)
        print("Name           :", self.User_name)
        print("Contact        :", self.contact)
        print("Email          :", self.email)
        print("Address        :", self.address)
        print("Total Purchase : Rs.", self.total_purchase)

    def PurchaseProduct(self):
        print(self.User_name, "purchased a product.")

    def ReturnProduct(self):
        print(self.User_name, "returned a product.")

    def ReplaceProduct(self):
        print(self.User_name, "replaced a product.")

    def GiveFeedback(self):
        print(self.User_name, "gave feedback.")